src/Elements/Input.react.js

```diff
@@ -181,25 +189,41 @@ class Input extends Component {
         let newValue = event.target.value;
 
         if (type === 'number') {
-            if (_.isEmpty(newValue)) {
-                if (required) {
-                    newValue = _.isNumber(min) ? min : (_.isNumber(max)? max : 0);
-                }
-            } else {
-                newValue = +newValue;
-                if (_.isNumber(max)) {
-                    newValue = Math.min(max, newValue);
+            if (this.inputTimer) {
+                clearTimeout(this.inputTimer);
+            }
+
+            this.inputTimer = setTimeout(() => {
+                if (_.isEmpty(newValue)) {
+                    if (required) {
+                        newValue = _.isNumber(min) ? min : (_.isNumber(max)? max : 0);
+                    }
+                } else {
+                    newValue = +newValue;
+                    if (_.isNumber(max)) {
+                        newValue = Math.min(max, newValue);
+                    }
+                    if (_.isNumber(min)) {
+                        newValue = Math.max(min, newValue);
+                    }
                 }
-                if (_.isNumber(min)) {
-                    newValue = Math.max(min, newValue);
+
+                if (_.isUndefined(onChange)) {
+                    this.setState({ value: newValue });
+                } else {
+                    onChange(newValue);
                 }
-            }
-        }
+            }, 500);
 
-        if (!_.isUndefined(onChange)) {
-            onChange(newValue);
+            if (_.isUndefined(onChange)) {
```

Comment:
Do we need lines 218-220 if we have 211-215?

Reply:
Yes, we do. Lines 218-220 immediately change the state, but lines 211-215 change the state by timer after the validation. These states can be different, and this is the main reason for using the timer - allows the user to quickly enter text (maybe not valid at the moment), but perform the validation after a pause.

Reply:
Sounds good and looks good. Thanks!

Reply:
I concur ... that was the _whole reason_ for this PR. :-)  It all has to do with the Min and Max Age inputs when setting Ministry or Serving Opportunity eligibility requirements.